internal/rules/resource_management.go

```diff
@@ -288,9 +327,10 @@ func (r ResourceManagementService) EnsureRoleBindings(
 
 // Helper methods for ResourceManagementService
 
-func (r ResourceManagementService) getContextWithTimeout(timeout time.Duration) (context.Context, context.CancelFunc) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	return ctx, cancel
+func (r ResourceManagementService) getContextWithTimeout(
```

Comment:
nit: this feels like a waste of lines of code just to change
`ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Second*30)`
to
`ctxWithTimeout, cancel := r.getContextWithTimeout(ctx, time.Second*30)`